### PR TITLE
Fixes #16063: Enable backtrace in relayd

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/16063

### CODE REVIEW

**Comment by amousset** [comment]
Commit modified

File: relay/sources/relayd/src/relayd.rs [hunk 1]
```diff
@@ -45,6 +45,14 @@ fn error_code(e: &Error) -> i32 {
 
 /// Everything in a lib to allow extensive testing
 fn main() {
+    // https://www.reddit.com/r/rust/comments/bnqina/why_does_not_rust_give_a_backtrace_by_default/
+    // https://internals.rust-lang.org/t/rust-backtrace-in-production-use/5609/2
+    // May be expensive only when backtraces are actually produced
+    // and can be helpful to troubleshoot production crashes
+    env::set_var("RUST_BACKTRACE", "1");
+
+    assert!(false);
```
Comment (VinceMacBuche) [inline relay/sources/relayd/src/relayd.rs]: Why do you need the assert ?

**Comment by amousset** [comment]
Commit modified

File: relay/sources/relayd/src/relayd.rs (hunk 1 above)
Comment (amousset) [reply relay/sources/relayd/src/relayd.rs]: To test that crashes actually produce backtraces :upside_down_face:

**Pull request merged**